test/e2eTests/extension.e2e.test.ts

```diff
@@ -74,6 +83,115 @@ describe('E2E Test', () => {
+                return Promise.race(
+                    [delay(timeOutMs, new TimeOutError('Timed out')),
+                    new Promise<void>(resolve => {
+                        (function check() {
+                            try {
+                                assertDiagnostics(diagnostics, true);
+                                resolve();
+                            } catch (err) {
+                                setTimeout(check, 1000);
+                            }
+                        })();
+                    })]
+                )
+            }
+
+            async function testOpenFile() {
+                checkDiagnostics({ [indexFileName]: [] });
+                const fileUri = Uri.file(path.join(vscode.workspace.workspaceFolders[0].uri.fsPath, "vscode-docs-build-e2e-test", indexFileName));
+                await vscode.window.showTextDocument(fileUri);
+                await checkDiagnostics({
+                    [indexFileName]: [
+                        <DiagnosticInfo>{
+                            range: new Range(7, 0, 7, 0),
+                            message: `Invalid file link: '${tempFileName}'.`,
+                            severity: vscode.DiagnosticSeverity.Warning,
+                            code: 'file-not-found',
+                        }
+                    ]
+                });
+            }
+
+            async function testModifyFile() {
+                const fileUri = Uri.file(path.join(vscode.workspace.workspaceFolders[0].uri.fsPath, "vscode-docs-build-e2e-test", indexFileName));
+                await vscode.window.showTextDocument(fileUri);
+                const workspaceEdit = new vscode.WorkspaceEdit();
+                workspaceEdit.replace(fileUri, new Range(new Position(1, 0), new Position(1, 5)), "titer");
+                await vscode.workspace.applyEdit(workspaceEdit);
+                await checkDiagnostics({
+                    [indexFileName]: [
+                        <DiagnosticInfo>{
+                            range: new Range(1, 0, 1, 0),
+                            message: `Missing required attribute: 'title'. Add a title string to show in search engine results.`,
+                            severity: vscode.DiagnosticSeverity.Warning,
+                            code: 'title-missing',
+                        },
+                        <DiagnosticInfo>{
+                            range: new Range(7, 0, 7, 0),
+                            message: `Invalid file link: '${tempFileName}'.`,
+                            severity: vscode.DiagnosticSeverity.Warning,
+                            code: 'file-not-found',
+                        }
+                    ]
+                });
+                workspaceEdit.replace(fileUri, new Range(new Position(1, 0), new Position(1, 5)), "title");
+                await vscode.workspace.applyEdit(workspaceEdit);
+            }
+
+            async function testCreateFile() {

```

Comment:
Since you have done some change in line#163, how could you make sure the expected diagnostics is caused by the file deletion?